tsdb/series_index.go

```diff
@@ -165,7 +165,11 @@ func (idx *SeriesIndex) GrowBy(delta int) {
 
 // Count returns the number of series in the index.
 func (idx *SeriesIndex) Count() uint64 {
-	return idx.OnDiskCount() + idx.InMemCount()
+	n := int64(idx.OnDiskCount()+idx.InMemCount()) - int64(len(idx.tombstones))
+	if n < 0 {
```

Comment:
This bothers me. How is this possible?

Reply:
It's not. We return a `uint64` though so if it does happen to cross into the negative threshold then we'll get stats at `math.UintMax`.